storage/mysql/log_storage_test.go

```diff
@@ -399,6 +401,7 @@ func TestDequeueLeaves(t *testing.T) {
 			t.Fatalf("Dequeued %d leaves but expected to get %d", len(leaves2), leavesToInsert)
 		}
 		ensureAllLeavesDistinct(leaves2, t)
+		ensureLeavesHaveQueueTimestamp(t, leaves2, fakeDequeueCutoffTime)
```

Comment:
Suggestion: since this is a new behavior we're adding I think it's worth having a separate test, instead appending the assert to all tests. We can use it to assert that dequeue works both with null and non-null timestamps. If the timestamp logic breaks then all tests here break; usually it's clearer to have a single, focused test break than the entire suite.

Reply:
done